predicators/teacher.py

```diff
@@ -187,9 +216,21 @@ class TeacherInteractionMonitor(utils.LoggingMonitor):
                                                  default_factory=list)
     _query_cost: float = field(init=False, default=0.0)
 
+    @property
+    def _teacher_env(self) -> BaseEnv:
+        """Exposes the teacher's environment to this monitor."""
+        # This is a temporary hack. When issue #1443 is resolved, this can be
+        # removed. That issue will require a fairly significant amount of work.
+        return self._teacher._env  # pylint: disable=protected-access
```

Comment:
Why did you change the `env` field of the Teacher class to be private (i.e, `_env`)? I don't see a great reason for it, especially given that you're accessing it like this

Reply:
It wasn't a field at all before (it was just a variable). I think we should leave it private because other classes shouldn't rely on it

Reply:
i see - makes sense!